File: LeetCode/Study Plan/Practice Problems/Easy/Root Equals Sum Of Children/Go/root_equal_sum_of_children.go
package main

import "fmt"

type TreeNode struct {
	Val int
	Left,
	Right *TreeNode
}

func New(val int) *TreeNode { return &TreeNode{Val: val, Left: nil, Right: nil} }

func root_equal_sum_of_children(root *TreeNode) bool {
	return root.Val == root.Left.Val+root.Right.Val
}

func main() {
	root := New(10)
	root.Left = New(4)
	root.Right = New(6)
	fmt.Println(root_equal_sum_of_children(root))
}

File: LeetCode/Study Plan/Practice Problems/Easy/Root Equals Sum Of Children/Java/rootEqualSumOfChildren.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum = 0;
    public boolean checkTree(TreeNode root) {
        recur(root);
        return root.val == sum;
    }

    public void recur(TreeNode root) {
        if (root == null) return;
        if (root.left == null && root.right == null) {
            sum += root.val;
        }

        recur(root.left);
        recur(root.right);
    }
}
